service/service.go

```diff
@@ -19,6 +20,25 @@ func logErr(l log.Logger, f func() error) {
 	}
 }
 
+func ContextAndMiddleware() (context.Context, server.Middleware, io.Closer, error) {
```

Comment:
@tnclong this is separating context middleware from HTTP serving, right? I've wanted this feature for a while 👏 👏 👏 

You can reimplement `ListenAndServe` to use this function, right?

Reply:
> You can reimplement ListenAndServe to use this function, right?

Fixed.

>  this is separating context middleware from HTTP serving, right?

Right.